metamorphic: TestMetaTwoInstance failed [data race, nil pointer]

https://github.com/cockroachdb/pebble/actions/runs/6819021314/job/18545683243?pr=3053

```
--- FAIL: TestMetaTwoInstance (120.61s)
    --- FAIL: TestMetaTwoInstance/execution (0.00s)
        --- FAIL: TestMetaTwoInstance/execution/random-018 (25.96s)
            meta.go:231: 
                ===== SEED =====
                1699575841871955433
                ===== ERR =====
                exit status 2
                ===== OUT =====
                ==================
                WARNING: DATA RACE
                Write at 0x00c0003fb3c0 by goroutine 58:
                  github.com/cockroachdb/pebble/metamorphic.(*test).clearObj()
                      /home/runner/work/pebble/pebble/metamorphic/test.go:308 +0x251
                  github.com/cockroachdb/pebble/metamorphic.(*closeOp).run()
                      /home/runner/work/pebble/pebble/metamorphic/ops.go:170 +0xe3
                  github.com/cockroachdb/pebble/metamorphic.RunOnce.func1()
                      /home/runner/work/pebble/pebble/metamorphic/meta.go:539 +0x24b
                  golang.org/x/sync/errgroup.(*Group).Go.func1()
                      /home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:75 +0x76
                
                Previous read at 0x00c0003fb3c0 by goroutine 59:
                  github.com/cockroachdb/pebble/metamorphic.(*test).getDB()
                      /home/runner/work/pebble/pebble/metamorphic/test.go:371 +0x84
                  github.com/cockroachdb/pebble/metamorphic.(*test).isFMV()
                      /home/runner/work/pebble/pebble/metamorphic/test.go:199 +0x33
                  github.com/cockroachdb/pebble/metamorphic.(*deleteOp).run()
                      /home/runner/work/pebble/pebble/metamorphic/ops.go:224 +0xf2
                  github.com/cockroachdb/pebble/metamorphic.RunOnce.func1()
                      /home/runner/work/pebble/pebble/metamorphic/meta.go:539 +0x24b
                  golang.org/x/sync/errgroup.(*Group).Go.func1()
                      /home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:75 +0x76
                
                Goroutine 58 (running) created at:
                  golang.org/x/sync/errgroup.(*Group).Go()
                      /home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:72 +0x124
                  github.com/cockroachdb/pebble/metamorphic.RunOnce()
                      /home/runner/work/pebble/pebble/metamorphic/meta.go:516 +0x1e6b
                  github.com/cockroachdb/pebble/internal/metamorphic.TestMetaTwoInstance()
                      /home/runner/work/pebble/pebble/internal/metamorphic/meta_test.go:86 +0x410
                  testing.tRunner()
                      /opt/hostedtoolcache/go/1.21.3/x64/src/testing/testing.go:1595 +0x2[38](https://github.com/cockroachdb/pebble/actions/runs/6819021314/job/18545683243?pr=3053#step:4:39)
                  testing.(*T).Run.func1()
                      /opt/hostedtoolcache/go/1.21.3/x64/src/testing/testing.go:1648 +0x44
                
                Goroutine 59 (running) created at:
                  golang.org/x/sync/errgroup.(*Group).Go()
                      /home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:72 +0x124
                  github.com/cockroachdb/pebble/metamorphic.RunOnce()
                      /home/runner/work/pebble/pebble/metamorphic/meta.go:516 +0x1e6b
                  github.com/cockroachdb/pebble/internal/metamorphic.TestMetaTwoInstance()
                      /home/runner/work/pebble/pebble/internal/metamorphic/meta_test.go:86 +0x410
                  testing.tRunner()
                      /opt/hostedtoolcache/go/1.21.3/x64/src/testing/testing.go:1595 +0x238
                  testing.(*T).Run.func1()
                      /opt/hostedtoolcache/go/1.21.3/x64/src/testing/testing.go:1648 +0x44
                ==================
                panic: runtime error: invalid memory address or nil pointer dereference
                [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe8b494]
                
                goroutine 74 [running]:
                github.com/cockroachdb/pebble.(*DB).FormatMajorVersion(...)
                	/home/runner/work/pebble/pebble/format_major_version.go:387
                github.com/cockroachdb/pebble/metamorphic.(*test).isFMV(0xc00020a960?, 0x20000029?, 0xf)
                	/home/runner/work/pebble/pebble/metamorphic/test.go:200 +0x34
                github.com/cockroachdb/pebble/metamorphic.(*deleteOp).run(0xc000434990, 0xc00020a960, {0xc000407160?, 0x0?})
                	/home/runner/work/pebble/pebble/metamorphic/ops.go:224 +0xf3
                github.com/cockroachdb/pebble/metamorphic.RunOnce.func1()
                	/home/runner/work/pebble/pebble/metamorphic/meta.go:5[39](https://github.com/cockroachdb/pebble/actions/runs/6819021314/job/18545683243?pr=3053#step:4:40) +0x24c
                golang.org/x/sync/errgroup.(*Group).Go.func1()
                	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:75 +0x77
                created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 21
                	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:72 +0x125
                
                
```

--- a/metamorphic/ops.go
+++ b/metamorphic/ops.go
@@ -614,7 +614,17 @@ func (o *ingestOp) receiver() objID { return o.dbID }
 func (o *ingestOp) syncObjs() objIDSlice {
 	// Ingest should not be concurrent with mutating the batches that will be
 	// ingested as sstables.
-	return o.batchIDs
+	objs := make([]objID, 0, len(o.batchIDs)+1)
+	objs = append(objs, o.batchIDs...)
+	addedDBs := make(map[objID]struct{})
+	for i := range o.derivedDBIDs {
+		_, ok := addedDBs[o.derivedDBIDs[i]]
+		if !ok && o.derivedDBIDs[i] != o.dbID {
+			objs = append(objs, o.derivedDBIDs[i])
+			addedDBs[o.derivedDBIDs[i]] = struct{}{}
+		}
+	}
+	return objs
 }
 
 func closeIters(

